platform/linuxbsd/freedesktop_portal_desktop.cpp

```diff
@@ -549,50 +536,18 @@ bool FreeDesktopPortalDesktop::color_picker(const String &p_xid, const Callable
 		append_dbus_dict_string(&arr_iter, "handle_token", token);
 		dbus_message_iter_close_container(&iter, &arr_iter);
 	}
-	DBusMessage *reply = dbus_connection_send_with_reply_and_block(monitor_connection, message, DBUS_TIMEOUT_INFINITE, &err);
-	dbus_message_unref(message);
 
-	if (!reply || dbus_error_is_set(&err)) {
-		ERR_PRINT(vformat("Failed to send DBus message: %s", err.message));
-		dbus_error_free(&err);
-		dbus_bus_remove_match(monitor_connection, cd.filter.utf8().get_data(), &err);
+	if (!send_request(message, token, cd.path, cd.filter)) {
 		return false;
 	}
 
-	// Update signal path.
-	{
-		DBusMessageIter iter;
-		if (dbus_message_iter_init(reply, &iter)) {
-			if (dbus_message_iter_get_arg_type(&iter) == DBUS_TYPE_OBJECT_PATH) {
-				const char *new_path = nullptr;
-				dbus_message_iter_get_basic(&iter, &new_path);
-				if (String::utf8(new_path) != path) {
-					dbus_bus_remove_match(monitor_connection, cd.filter.utf8().get_data(), &err);
-					if (dbus_error_is_set(&err)) {
-						ERR_PRINT(vformat("Failed to remove DBus match: %s", err.message));
-						dbus_error_free(&err);
-						return false;
-					}
-					cd.filter = String::utf8(new_path);
-					dbus_bus_add_match(monitor_connection, cd.filter.utf8().get_data(), &err);
-					if (dbus_error_is_set(&err)) {
-						ERR_PRINT(vformat("Failed to add DBus match: %s", err.message));
-						dbus_error_free(&err);
-						return false;
-					}
-				}
-			}
-		}
-	}
-	dbus_message_unref(reply);
-
 	MutexLock lock(color_picker_mutex);
 	color_pickers.push_back(cd);
 
 	return true;
 }
 
-bool FreeDesktopPortalDesktop::_is_interface_supported(const char *p_iface) {
+bool FreeDesktopPortalDesktop::_is_interface_supported(const char *p_iface, const uint32_t p_minimum_version) {
```

Comment:
```suggestion
bool FreeDesktopPortalDesktop::_is_interface_supported(const char *p_iface, uint32_t p_minimum_version) {
```
You missed this part